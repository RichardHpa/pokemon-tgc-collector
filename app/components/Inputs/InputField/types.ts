type inputTypes = "text" | "email" | "password" | "number" | "tel" | "url";

export type InputFieldProps = {
  className?: string;
  disabled?: boolean;
  error?: string | null;
  label: string;
  type?: inputTypes;
  autoComplete?: string;
} & React.ComponentPropsWithoutRef<"input">;
