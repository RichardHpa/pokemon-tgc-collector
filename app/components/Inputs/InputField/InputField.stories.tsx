import { InputField as InputFieldComponent } from "./InputField";

import type { ComponentStory, ComponentMeta } from "@storybook/react";

const Template: ComponentStory<typeof InputFieldComponent> = ({ ...args }) => {
  return <InputFieldComponent {...args} label="Input Field Label" />;
};

export const InputField = Template.bind({});

export default {
  title: "Components/Inputs/InputField",
  component: InputFieldComponent,
} as ComponentMeta<typeof InputFieldComponent>;
