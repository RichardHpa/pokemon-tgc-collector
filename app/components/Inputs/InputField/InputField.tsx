import { forwardRef } from "react";

import type { InputFieldProps } from "./types";

export const InputField = forwardRef<HTMLInputElement, InputFieldProps>(
  (
    {
      disabled = false,
      error,
      label,
      type = "text",
      name,
      autoComplete = "off",
      required = false,
      ...props
    },
    ref
  ) => {
    return (
      <div>
        <label
          htmlFor={name}
          className="block text-sm font-medium text-gray-700 dark:text-gray-300"
        >
          {label} {required && <span>*</span>}
        </label>
        <div className="mt-1">
          <input
            {...props}
            id={name}
            ref={ref}
            name={name}
            type={type}
            required={required}
            autoComplete={autoComplete}
            aria-invalid={error ? true : undefined}
            aria-describedby={`${name}-error`}
            className="block w-full rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-900 focus:border-blue-500 focus:ring-blue-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-blue-50 dark:focus:ring-blue-500"
            disabled={disabled}
          />
          {error && (
            <div className="pt-1 text-red-700" id={`${name}-error`}>
              {error}
            </div>
          )}
        </div>
      </div>
    );
  }
);

InputField.displayName = "InputField";
