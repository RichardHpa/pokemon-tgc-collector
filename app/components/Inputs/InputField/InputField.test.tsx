import { render, screen } from "@testing-library/react";

import { withMarkup } from "test/helpers/withMarkup";

import { InputField } from "./InputField";

describe("InputField", () => {
  test("renders InputField", () => {
    render(<InputField label="text field" />);
    expect(screen.getByRole("textbox")).toBeInTheDocument();
  });

  test("renders InputField with label", () => {
    render(<InputField label="text field" />);
    expect(screen.getByText("text field")).toBeInTheDocument();
  });

  test("renders InputField with required", () => {
    const { getByText } = render(<InputField label="text field" required />);
    const getByTextWithMarkup = withMarkup(getByText);
    getByTextWithMarkup("text field *");
  });

  test("renders InputField with error", () => {
    render(<InputField label="text field" error="error" />);
    expect(screen.getByText("error")).toBeInTheDocument();
  });

  test("renders InputField snapshot", () => {
    const { container } = render(<InputField label="text field" />);
    expect(container).toMatchSnapshot();
  });
});
