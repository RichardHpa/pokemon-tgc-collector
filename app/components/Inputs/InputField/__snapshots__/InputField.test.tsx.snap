// Vitest Snapshot v1

exports[`InputField > renders InputField snapshot 1`] = `
<div>
  <div>
    <label
      class="block text-sm font-medium text-gray-700 dark:text-gray-300"
      for="text-field"
    >
      text field
       
    </label>
    <div
      class="mt-1"
    >
      <input
        aria-describedby="text-field-error"
        autocomplete="off"
        class="block w-full rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-900 focus:border-blue-500 focus:ring-blue-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-blue-50 dark:focus:ring-blue-500"
        id="text-field"
        name="text-field"
        type="text"
      />
    </div>
  </div>
</div>
`;
